lib/yaml-reader.js

```diff
@@ -90,28 +91,40 @@ function YamlReader(options) {
     cb(null, path.resolve(folderPath, fileName));
   };
 
-  this.readFile = function(filePath, cb) {
-    fs.readFile(filePath, {encoding: 'utf-8'}, function(err, data){
-      if(err)
-        return void cb(err);
-      cb(null, data);
-    });
+  this.findFileAbsolutePaths = function(globSearch, cb) {
+    var folderPath = self.options.folderPath;
+    glob(path.resolve(folderPath, globSearch), cb);
   };
-  
+
+  this.readFiles = function(filePaths, cb) {
+    if (!Array.isArray(filePaths)) { filePaths = [filePaths]; }
+    async.parallel(filePaths.map(function (filePath) {
+      return function readYamlFile(fileCallback) {
+        fs.readFile(filePath, {encoding: 'utf-8'}, fileCallback);
+      };
+    }), cb);
+  };
+
   this.readFileSafe = function(filePath, cb) {
-    self.readFile(filePath, function(err, data){
+    self.readFiles(filePath, function(err, data){
       if(err)
         return void cb(null, null);
       cb(null, data);
     });
   };
 
-  this.parseYamlToJsObject = function(yamlString, cb) {
-    try {
-      cb(null, yaml.safeLoad(yamlString));
-    } catch (e) {
-      cb(e);
+  this.parseYamlToJsObject = function(yamlStrings, cb) {
+    if (!Array.isArray(yamlStrings)) { yamlStrings = [yamlStrings]; }
+    var merged = {};
+    for(var i =0; i < yamlStrings.length; i++) {
+      try {
+        var parsed = yaml.safeLoad(yamlStrings[i]);
+        merged = _.assign(merged, parsed);
```

Comment:
Should we be signalling an error if we have two different values for the same config item? It seems like this could make an error adding configs cause indeterminate behaviour, where sometimes one config value was read, and sometimes another.